server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/VacationMailet.java

```diff
@@ -77,20 +82,36 @@ public void service(Mail mail) {
     private boolean shouldSendNotification(Vacation vacation, Mail processedMail, MailAddress recipient, ZonedDateTime processingDate) {
         try {
             return vacation.isActiveAtDate(processingDate)
-                && ! automaticallySentMailDetector.isAutomaticallySent(processedMail);
+                && ! automaticallySentMailDetector.isAutomaticallySent(processedMail)
+                && hasNotSentNotificationsYet(processedMail, recipient);
         } catch (MessagingException e) {
             LOGGER.warn("Failed detect automatic response in a mail from {} to {}", processedMail.getSender(), recipient, e);
             return false;
         }
     }
 
+    private boolean hasNotSentNotificationsYet(Mail processedMail, MailAddress recipient) {
+        CompletableFuture<Boolean> hasAlreadyBeenSent = notificationRegistry.isRegistered(
+            AccountId.fromString(recipient.toString()),
+            RecipientId.fromMailAddress(processedMail.getSender()));
+        try {
+            return !hasAlreadyBeenSent.join();
+        } catch (Throwable t) {
```

Comment:
I am expecting to catch failures to see if we have sent a notification to a recipient.
Especially a Cassandra exception
Here we know the vacation is activated, the mail not sent automatically.
The question is "should I send a notification ? or shouldn't I?"
I answer by saying "Cassandra failures are rare enough so that it do not floud peoples. We don't want the vacation feature to be done for this". Hence me managing the exception this way.

You have the associated tests in VacationMailetTest